message/router.go

```diff
@@ -135,7 +147,10 @@ func (r *Router) Logger() watermill.LoggerAdapter {
 func (r *Router) AddMiddleware(m ...HandlerMiddleware) {
 	r.logger.Debug("Adding middlewares", watermill.LogFields{"count": fmt.Sprintf("%d", len(m))})
 
-	r.middlewares = append(r.middlewares, m...)
+	for _, handlerMiddleware := range m {
+		middleware := NewMiddleware(handlerMiddleware, "")
```

Comment:
We don't enforce at the moment that the handler name is not empty, won't this be confusing if someone would create handler with empty name? Maybe it would be slightly cleaner to introduce some explicit boolean flag on the handler struct for this purpose.